### PR TITLE
Add json serialization to sweeps

### PR DESCRIPTION
**Pull request by dabacon**
Adds missing json serialization for sweep objects.

### CODE REVIEW

File: cirq-core/cirq/study/sweeps.py [hunk 1]
```diff
@@ -196,6 +197,9 @@ def param_tuples(self) -> Iterator[Params]:
     def __repr__(self) -> str:
         return 'cirq.UnitSweep'
 
+    def _json_dict_(self) -> Dict[str, Any]:
```
Comment (95-martin-orion) [inline cirq-core/cirq/study/sweeps.py:200]: Based on the name, `_Unit` is a private class. Do we want it to be serializable?

If so, I think this needs a `_from_json_dict_` classmethod.

**Review by 95-martin-orion — APPROVED** [review]
LGTM (% one concern, below). What is the expected use case for this?

**Comment by dabacon** [comment]
Use case is that objects that set our context for experiments should be serializable.

File: cirq-core/cirq/study/sweeps.py (hunk 1 above)
Comment (95-martin-orion) [reply cirq-core/cirq/study/sweeps.py:200]: SGTM. This can merge once the tests are sorted out.

**Comment by dabacon** [comment]
Ugh bad merge fixing

**Pull request merged**